cadc-storage-adapter-fs/src/intTest/java/org/opencadc/inventory/storage/fs/OpaquePutTxnTest.java

```diff
@@ -0,0 +1,144 @@
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.inventory.storage.fs;
+
+import ca.nrc.cadc.util.Log4jInit;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.opencadc.inventory.storage.InvalidConfigException;
+import org.opencadc.inventory.storage.test.StorageAdapterPutTxnTest;
+
+/**
+ *
+ * @author pdowler
+ */
+public class OpaquePutTxnTest extends StorageAdapterPutTxnTest {
+    private static final Logger log = Logger.getLogger(OpaquePutTxnTest.class);
+
+    static final int BUCKET_LEN = 2;
+    static final File ROOT_DIR;
+    
+    static {
+        Log4jInit.setLevel("org.opencadc.inventory.storage", Level.INFO);
+        ROOT_DIR = new File("build/tmp/opaque-int-tests");
+        ROOT_DIR.mkdir();
+    }
+    
+    final OpaqueFileSystemStorageAdapter ofsAdapter;
+            
+    public OpaquePutTxnTest() throws InvalidConfigException {
+        super(new OpaqueFileSystemStorageAdapter(ROOT_DIR, BUCKET_LEN));
+        this.ofsAdapter = (OpaqueFileSystemStorageAdapter) super.adapter;
+
+        log.debug("    content path: " + ofsAdapter.contentPath);
+        log.debug("transaction path: " + ofsAdapter.txnPath);
+        Assert.assertTrue("testInit: contentPath", Files.exists(ofsAdapter.contentPath));
+        Assert.assertTrue("testInit: txnPath", Files.exists(ofsAdapter.txnPath));
+    }
+    
+    @Before
+    public void cleanupBefore() throws IOException {
+        log.info("cleanupBefore: " + ofsAdapter.contentPath.getParent());
+        if (Files.exists(ofsAdapter.contentPath)) {
+            Files.walkFileTree(ofsAdapter.contentPath, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    Files.delete(file);
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                    if (!ofsAdapter.contentPath.equals(dir)) {
+                        Files.delete(dir);
```

Comment:
since this is cleanup before test, it always nukes the directories as well in case someone wants to test with different test setup (eg change the bucketLength from 1)